allspice/utils/list_components.py

```diff
@@ -175,14 +182,39 @@ def list_components_for_altium(
     schdoc_files_in_proj = _extract_schdoc_list_from_prjpcb(prjpcb_ini)
     allspice_client.logger.info("Found %d SchDoc files", len(schdoc_files_in_proj))
 
-    schdoc_jsons = {
-        schdoc_file: _fetch_generated_json(
-            repository,
-            _resolve_prjpcb_relative_path(schdoc_file, prjpcb_file),
-            ref,
-        )
-        for schdoc_file in schdoc_files_in_proj
-    }
+    tree = repository.get_tree(recursive=True)
+    files_in_this_repo = [file.path for file in tree]
+
+    schdoc_jsons = {}
+
+    for schdoc_file in schdoc_files_in_proj:
```

Comment:
I don't think we need to go through all files here. We know which sheets are design sheets because the section they come from is different, so we can only do this work for the design sheets, and save ourselves this work for the regular schematics.